src/query/interpret/eval.hpp

```diff
@@ -159,50 +224,53 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
   }
 
   TypedValue Visit(SubscriptOperator &list_indexing) override {
-    auto lhs = list_indexing.expression1_->Accept(*this);
+    ReferenceExpressionEvaluator referenceExpressionEvaluator(frame_, symbol_table_, ctx_);
+
+    TypedValue *lhs_ptr = list_indexing.expression1_->Accept(referenceExpressionEvaluator);
+    TypedValue lhs;
+    bool referenced = nullptr != lhs_ptr;
+    if (nullptr == lhs_ptr) {
```

Comment:
```suggestion
    const auto referenced = nullptr != lhs_ptr;
    if (!referenced) {
```

I didn't see that you used `referenced` later on.